speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DataCompressionMenu.cpp

```diff
@@ -0,0 +1,239 @@
+#include <tgfclient.h>
+#include "guimenu.h"
+#include "legacymenu.h"
+#include "Mediator.h"
+#include "ConfigEnums.h"
+#include "DataCompressionMenu.h"
+
+// Parameters used in the xml files
+#define PRM_DATA_COMPRESSION   "DataCompressionRadioButtonList"
+#define PRM_CUSTOM_COMPRESSION "CustomCompressionEdit"
+
+// Names for the config file
+#define COMPRESSION_FILEPATH    "config/DataCompressionMenu.xml"
+#define COMPRESSION_SCREEN_NAME "DataCompressionMenu"
+
+#define MAX_COMPRESSION_LEVEL 49
+
+// GUI screen handles
+static void* s_scrHandle = nullptr;
+static void* s_prevHandle = nullptr;
+
+// GUI settings Id's
+int m_compressionButtonList;
+
+// InterventionType
+DataCompressionType m_dataCompressionType;
```

Comment:
This parameter is only set but never used in logic (except for synching radiobuttons, etc). I think you used the wrong variable in your switch statement further down.

Reply:
You are right. Did changing the switch statement solve this issue?